### ISSUE TITLE
write_disk.frame fails on simple example

### ISSUE THREAD

**Issue by kendonB** [xiaodaigh/disk.frame#153]
```` r
library(disk.frame)
write_disk.frame(iris)
#> Error in `[.data.frame`(df, , {: object '.out.disk.frame.id' not found

iris_df = as.disk.frame(iris)
write_disk.frame(iris_df)
#> Error in overwrite_check(outdir, overwrite): overwrite  = FALSE and outdir 'C:\Users\kmbel\AppData\Local\Temp\RtmpUn1tcG\file16b02aab2791.df' already exists
````

<sup>Created on 2019-09-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by xiaodaigh** [comment]
The first error is the intended behaviour, see the documentation for `write_disk.frame`:

> if df is a data.frame, then df must contain the column .out.disk.frame.id. This is intended to be a low-level version of writing disk.frames. Using the as.disk.frame function is recommended for most cases

I realise now that the default behaviour here is super confusing. Because the way you are using it seem more natural. I will adjust the API for this. This is super user-unfriendly. 

The second one is also the intended behaviour, because by default `overwrite = FALSE`, so you need ```r
iris_df = as.disk.frame(iris)
write_disk.frame(iris_df, overwrite = TRUE)
```

And `overwrite` is tricky. Generally, you only use disk.frame if the data is large, so if `overwrite = TRUE` by default, then you could be inadvently overwriting a large disk.frame that took many minutes to create. But I need to think about the overwrite behaviour and maybe make it default to TRUE or by a user defined option.

**Comment by xiaodaigh** [comment]
Although, the error message could be more helpful.

**Comment by kendonB** [comment]
Thanks for this clarification. I still think there's a bug in the second case as you get the error even when the folder is new:

```` r
library(disk.frame)
iris_df = as.disk.frame(iris)
write_disk.frame(iris_df, outdir = "new_dir")
#> Error in overwrite_check(outdir, overwrite): overwrite  = FALSE and outdir 'new_dir' already exists
````

<sup>Created on 2019-09-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by xiaodaigh** [comment]
> Thanks for this clarification. I still think there's a bug in the second case as you get the error even when the folder is new:

This is where the error message could be more helpful.

**Comment by xiaodaigh** [comment]
> Thanks for this clarification. I still think there's a bug in the second case as you get the error even when the folder is new:

Oh! You are right.

I've fixed both. They should work now! Fixed by #156

```r
#devtools::install_github("xiaodaigh/disk.frame")
library(disk.frame)
setup_disk.frame()

write_disk.frame(iris) # works

iris_df = as.disk.frame(iris)
write_disk.frame(iris_df) # works
```
